pkg/cli/mirror/fbc_operators_test.go

```diff
@@ -326,19 +265,69 @@ func TestGetRelatedImages(t *testing.T) {
 			},
 			err: "",
 		},
+		{
+			desc:        "multi arch nominal case passes",
+			configsPath: multiTestData,
+			packages: []v1alpha2.IncludePackage{
+				{
+					Name: "node-observability-operator",
+				},
+			},
+			expectedRelatedImages: []declcfg.RelatedImage{
+				{
+					Image: "registry.redhat.io/noo/node-observability-operator-bundle-rhel8@sha256:25b8e1c8ed635364d4dcba7814ad504570b1c6053d287ab7e26c8d6a97ae3f6a",
+					Name:  "node-observability-operator",
+				},
+				{
+					Image: "registry.redhat.io/openshift4/ose-kube-rbac-proxy@sha256:bb54bc66185afa09853744545d52ea22f88b67756233a47b9f808fe59cda925e",
+					Name:  "kube-rbac-proxy",
+				},
+				{
+					Name:  "manager",
+					Image: "registry.redhat.io/noo/node-observability-rhel8-operator@sha256:0040925e971e4bb3ac34278c3fb5c1325367fe41ad73641e6502ec2104bc4e19",
+				},
+				{
+					Name:  "agent",
+					Image: "registry.redhat.io/noo/node-observability-agent-rhel8@sha256:59bd5b8cefae5d5769d33dafcaff083b583a552e1df61194a3cc078b75cb1fdc",
+				},
+			},
+			err: "",
+		},
 	}
 	for _, c := range cases {
 		t.Run(c.desc, func(t *testing.T) {
-			//Untar the configs blob to tmpdir
-			stream, err := os.Open(c.configsPath)
-			if err != nil {
-				t.Fatalf("unable to open %s: %v", c.configsPath, err)
+			var actualDir string
+			// handle images... all images are expected to have the same content in these tests
+			// for multi arch, the actualDir gets set twice, but its the same content so it does not matter
+			handleImage := func(t *testing.T, img v1.Image) {
+				t.Helper()
+				tmpDir := t.TempDir()
+				dir, err := extractDeclarativeConfigFromImage(img, tmpDir)
+				require.NoError(t, err)
+				actualDir = dir
 			}
-			err = UntarLayers(stream, tmpdir, "configs/")
-			if err != nil {
-				t.Fatalf("unable to untar %s: %v", c.configsPath, err)
+			// recursive function to handle image indexes
+			var handleIndex func(t *testing.T, idx v1.ImageIndex)
+			handleIndex = func(t *testing.T, idx v1.ImageIndex) {
+				t.Helper()
+				idxManifest, err := idx.IndexManifest()
+				require.NoError(t, err)
+				for _, descriptor := range idxManifest.Manifests {
+					if descriptor.MediaType.IsImage() {
+						img, err := idx.Image(descriptor.Digest)
+						require.NoError(t, err)
+						handleImage(t, img)
+					} else if descriptor.MediaType.IsIndex() {
+						innerIdx, err := idx.ImageIndex(descriptor.Digest)
+						require.NoError(t, err)
+						handleIndex(t, innerIdx)
+					}
+				}
 			}
-			relatedImages, err := getRelatedImages(filepath.Join(tmpdir, "configs"), c.packages)
+			imageIndex, err := c.configsPath.ImageIndex()
+			require.NoError(t, err)
+			handleIndex(t, imageIndex)
+			relatedImages, err := getRelatedImages(actualDir, c.packages)
```

Comment:
This is the part that PR #627 OCPBUGS-13116 is also changing.
getRelatedImages should be called on the declarativeConfig that is already filtered by a previous call to render. Therefore, there is no need to start back from the image.

Reply:
thanks for the heads up. I've rebased my PR on those changes.